src/Metadata/Resource/ResourceNameCollection.php

```diff
@@ -19,13 +19,15 @@
 final class ResourceNameCollection implements \IteratorAggregate, \Countable
 {
     private $classes;
+    private $context;
 
     /**
      * @param string[] $classes
      */
-    public function __construct(array $classes = [])
+    public function __construct(array $classes = [], array $context = [])
```

Comment:
`YamlResourceNameCollectionFactor`y gets `paths` and returns the `ResourceNameCollection`. So, I'm parsing relevant configurations (`paths`) to get the `resourceClass`.

`YamlResourceName` has to get the configuration for this node, but actually the `YamlResourceName` does not have any informations about which configuration file it came from, nor what values were associated to the class. 

So, I want to find my original "raw" configuration in the `YamlResourceMetadataFactory` too without parsing every path to find the configuration corresponding to the `resourceClass`.

It's different from caching, because we're already caching `ResourceMetadata` and we don't really care about the "raw" configuration of a resource.